src/components/DetailsMetrics/DetailsMetrics.js

```diff
@@ -17,17 +17,110 @@ illegal under applicable law, and the grant of the foregoing license
 under the Apache 2.0 license is conditioned upon your compliance with
 such restriction.
 */
-import React, { useEffect, useState } from 'react'
-import { useSelector } from 'react-redux'
-import { useDispatch } from 'react-redux'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
 import { isEmpty } from 'lodash'
 
+import GenericMetricChart from '../MetricChart/MetricChart'
+import StatsCard from '../../common/StatsCard/StatsCard'
+
 import detailsActions from '../../actions/details'
+import { groupMetricByApplication } from '../../elements/MetricsSelector/metricsSelector.utils'
+import { getBarChartMetricConfig, getLineChartMetricConfig } from '../../utils/getMetricChartConfig'
+
+import { TextTooltipTemplate, Tooltip } from 'iguazio.dashboard-react-controls/dist/components'
+import { ReactComponent as NoData } from 'igz-controls/images/no-data-metric-icon.svg'
+import { ReactComponent as Metrics } from 'igz-controls/images/metrics-icon.svg'
+
+import './DetailsMetrics.scss'
 
 const DetailsMetrics = ({ selectedItem }) => {
   const [metrics, setMetrics] = useState([])
   const detailsStore = useSelector(store => store.detailsStore)
   const dispatch = useDispatch()
+  const lineConfig = useMemo(() => getLineChartMetricConfig(), [])
+  const barConfig = useMemo(() => getBarChartMetricConfig(), [])
+
+  // TODO: refactor the calculateHistogram
+  const calculateHistogram = useCallback((data, item) => {
+    const numberOfBins = 5
+    const min = Math.min(...data)
+    const max = Math.max(...data)
+    const range = max - min === 0 ? 1 : max - min
+    const binSize = range / numberOfBins
+
+    const bins = Array(numberOfBins).fill(0)
+    data.forEach(value => {
+      const binIndex = Math.min(Math.floor((value - min) / binSize), numberOfBins - 1)
+      bins[binIndex]++
+    })
+
+    const totalCount = data.length
+    const binPercentages = bins.map(count => ((count / totalCount) * 100).toFixed(1))
+    const binLabels = Array.from({ length: numberOfBins }, (_, i) => {
+      const rangeStart = (min + i * binSize).toFixed(2)
+      const rangeEnd =
+        i === numberOfBins - 1 ? max.toFixed(2) : (min + (i + 1) * binSize).toFixed(2)
+      return `${rangeStart} - ${rangeEnd}`
+    })
+
+    const calculateAverages = data => {
+      return data.map(item => {
+        if (max === min) return max
+        const [num1, num2] = item.split(' - ').map(parseFloat)
+        const average = (num1 + num2) / 2
+        return (Math.abs(average * 100) / 100).toFixed(1)
+      })
+    }
+
+    let averageValue = calculateAverages(binLabels)
+    const adjustArray = (binPercentages, averageValue) => {
+      return binPercentages.map((value, index) => {
+        return parseFloat(averageValue[index]) !== 0 ? value : ''
+      })
+    }
+    if (max === min) {
+      averageValue = adjustArray(averageValue, binPercentages)
+    }
```

Comment:
format the code. Add empty lines where needed. Use meaningful variables instead of data, item, min, max, etc.

Reply:
done